WordPress/Classes/Services/NullBlogPropertySanitizer.swift

```diff
@@ -0,0 +1,66 @@
+import Foundation
+
+/// Delete invalid rows in the database whose required blog properties are NULL
+///
+/// This is born from the investigation of the crash described in #12028 (https://git.io/JePdZ).
+/// Unfortunately, we could not find the cause of it. The findings are described in that issue.
+///
+/// This tries to “fix” the issue by deleting the orphaned entities like `Post` whose `blog` were
+/// set to NULL. They are currently inaccessible because they are not attached to any `Blog`. But
+/// leaving them in the database would cause a crash if Core Data tries to save new entities.
+///
+@objc class NullBlogPropertySanitizer: NSObject {
+    private let store: UserDefaults
+    private let key = "null-property-sanitization"
+
+    private let context: NSManagedObjectContext
+
+    @objc init(store: UserDefaults = UserDefaults.standard,
```

Comment:
I see that you are passing the `UserDefaults` here for testing, you might want to take a look at the existing `KeyValueDatabase` protocol which does a similar thing.

Reply:
I tried that!

But since we're exposing that to ObjC I'm getting some weird errors when adding `@objc` in the `KeyValueDatabase`. :(

Reply:
Gotcha, I think what can work in that case is keeping the current initializer as Swift only, then adding a convenience `@objc init()` that calls the designated one with the default values, rather than using parameter defaults.

Reply:
Done from [ef80779 to 6057324](https://github.com/wordpress-mobile/WordPress-iOS/pull/13021/files/ef807792ff823cf747bfd7200126189aceca803c..605732455758ee30d00a82d9868953d007127bee).